pkg/server/apis/v1alpha1/types.go

```diff
@@ -248,3 +248,40 @@ type StorageUsage struct {
 type StorageCleanup struct {
 	Paths []string `json:"paths"`
 }
+
+// ExecutionContextSpec describes the execution context
+type ExecutionContextSpec struct {
+	Cluster     string `json:"cluster"`
+	Namespace   string `json:"namespace"`
+	Integration string `json:"integration"`
+	PVC         string `json:"pvc"`
+}
+
+// ExecutionContextStatus describe the status of execution context, it contains information that affects
+// pipeline execution, like reserved resources, pvc status.
+type ExecutionContextStatus struct {
+	// Phase of the execution context, could be 'Ready', 'NotReady' or 'Unknown'
+	Phase             ExecutionContextPhase                `json:"phase"`
+	ReservedResources map[core_v1.ResourceName]string      `json:"reservedResources"`
```

Comment:
You can comment what is ReservedResources with examples.

Reply:
OK